login-workflow/src/new-architecture/components/WorkflowSecureTextField.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { ReactNode, useState } from 'react';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import InputAdornment from '@mui/material/InputAdornment';
+import IconButton from '@mui/material/IconButton';
+import Visibility from '@mui/icons-material/Visibility';
+import VisibilityOff from '@mui/icons-material/VisibilityOff';
+
+/**
+ * Component that renders textfield with a visibility toggle. The toggle changes the
+ * input from hidden to visible.
+ *
+ * @param props all props will be passed to the underlying TextField component
+ *
+ * @category Component
+ */
+export const WorkflowSecureTextField: React.FC<
+    React.PropsWithChildren<React.PropsWithChildren<TextFieldProps>> & { icon?: ReactNode }
+> = (props) => {
```

Comment:
This component is already in the new-architecture as PasswordTextField. Use that one instead of creating this new one.